puzzles/sequential-digits/sequential-digits.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+func leftmostDigit(n int) int {
+	for n >= 10 {
+		n = n / 10
+	}
+
+	return n
+}
+
+func numDigits(n int) int {
+	var count int = 0
+	for n > 0 {
+		n = n / 10
+		count = count + 1
+	}
+	return count
+}
+
+// assumes leftDigit and E are valid constraints for producing a sequential number, per validSequentialNumberConstraints
+func getSequentialNumber(leftDigit int, E int) int {
+	if leftDigit > 9 || E < 0 {
+		return 0
+	}
+
+	var n int = leftDigit * int(math.Pow(10, float64(E)))
+	return n + getSequentialNumber(leftDigit+1, E-1)
+}
+
+func validSequentialNumberConstraints(leftDigit int, E int) bool {
+	return leftDigit+E <= 9
+}
+
+// https://leetcode.com/problems/sequential-digits/
+// An integer has sequential digits if and only if each digit in the number is one more than the previous digit.
+// Return a sorted list of all the integers in the range [low, high] inclusive that have sequential digits.
+func sequentialDigits(low int, high int) []int {
+	var startingDigit int = leftmostDigit(low)
+	var startingE int = numDigits(low) - 1 // e.g. if low is in range 10 - 99, E is 1
+	var maxE int = numDigits(high)
```

Comment:
i think this could also be subtracted by one, saving a loop iteration downstream  
```suggestion
	var maxE int = numDigits(high) - 1
```

Reply:
validated with test cases this is correct